packages/components/src/components/notification/notification.ts

```diff
@@ -0,0 +1,339 @@
+          if (toastStack.querySelector('sd-notification') === null) {
+            toastStack.remove();
+          }
+        },
+        { once: true }
+      );
+    });
+  }
+
+  render() {
+    const variantToName = {
+      info: 'info',
+      success: 'success',
+      warning: 'warning',
+      error: 'error'
+    };
+
+    return html`
+      <div
+        part="base"
+        class=${cx('w-full overflow-hidden flex items-stretch relative m-2')}
+        role="alert"
+        id="notification"
+        aria-hidden=${this.open ? 'false' : 'true'}
+        @mouseenter=${this.onHover}
+        @mouseleave=${this.onHoverEnd}
+      >
+        <slot
+          name="icon"
+          part="icon"
+          class=${cx(
+            'min-w-min flex items-center px-3 justify-center',
+            {
+              info: 'bg-info',
+              success: 'bg-success',
+              warning: 'bg-warning',
+              error: 'bg-error'
+            }[this.variant]
+          )}
+        >
+          <sd-icon name=${variantToName[this.variant] || ''} library="system" class="h-6 w-6 text-white"></sd-icon>
+        </slot>
+
+        <div
+          part="content"
+          class=${cx(
+            'h-full w-full p-1 gap-2 flex items-center justify-stretch bg-white',
+            'border-solid border-[1px] border-l-0 border-neutral-400'
+          )}
+        >
+          <slot part="message" class="block w-full pl-3 py-2" aria-live="polite"></slot>
+
+          ${this.closable
+            ? html`
+                <sd-button
+                  size="md"
+                  variant="tertiary"
+                  part="close-button"
+                  class="ml-auto flex flex-[0_0_auto] items-stretch"
+                  label=${this.localize.term('close')}
+                  @click=${this.handleCloseClick}
+                >
+                  <sd-icon name="close" library="system" color="currentColor"></sd-icon>
+                </sd-button>
+              `
+            : ''}
+        </div>
+
+        ${this.durationIndicator
+          ? html`
+              <div
+                part="duration-indicator-current"
+                style=${`animation-duration: ${this.duration}ms`}
+                class=${cx(`h-[2px] bottom-0 absolute bg-primary z-10 width-animation`)}
+              ></div>
+              <div part="duration-indicator-total" class="w-full h-[2px] bottom-0 absolute bg-neutral-400"></div>
+            `
+          : ''}
+      </div>
+    `;
+  }
+
+  /**
+   * Inherits Tailwindclasses and includes additional styling.
+   */
+  static styles = [
+    componentStyles,
+    SolidElement.styles,
+    css`
+      :host {
+        display: contents;
+      }
+
+      .width-animation {
+        width: 0%;
+        animation: grow;
+        animation-timing-function: linear;
+        animation-delay: 0s;
+      }
+
+      #notification:hover .width-animation {
+        animation-play-state: paused;
+      }
```

Comment:
```suggestion
      .width-animation {
        width: 0%;
        animation: grow;
        animation-timing-function: linear;
        animation-delay: 0s;
      }

      #notification:hover .width-animation {
        animation-play-state: paused;
      }
```

https://tailwindcss.com/docs/animation

Can you make a shorthand version of that? And please call it `animate-grow` or similar. And can you set w-0 ?

Reply:
I moved the core of the animation to tailwind.config. But, I excluded the duration since that depends on an attribute.